### PR TITLE
ENH: Don't enforce BIDS & Raw channel name equivalence in read_raw_bids()

### LINKED ISSUE #697: write_raw_bids() creates BIDS datasets that read_raw_bids() refuses to read if channels were renamed before writing
### Problem description
@MaelysSolal ran into this.

Yesterday, we converted a dataset to BIDS after renaming some channels.. Today, the same data cannot be read.

### Reproduction
MWE:

```python
import mne
from mne.datasets import eegbci

from mne_bids import read_raw_bids, write_raw_bids, BIDSPath


subject = '001'
task = 'RestEyesClosed'
bids_root = '/tmp/bci-bids'
bids_path = BIDSPath(subject=subject, task=task, root=bids_root)

raw_fname = eegbci.load_data(subject=1, runs=2, update_path=True)[0]
raw = mne.io.read_raw_edf(raw_fname)
raw.info['line_freq'] = 50

raw.rename_channels({'Fc5.': 'FC5'})  # <--- !!!

write_raw_bids(raw=raw, bids_path=bids_path, overwrite=True)
read_raw_bids(bids_path=bids_path)
```
Output:

> RuntimeError                              Traceback (most recent call last)
> /private/tmp/mwe-rename-chs.py in 
>      18 
>      19 write_raw_bids(raw=raw, bids_path=bids_path, overwrite=True)
> ---> 20 read_raw_bids(bids_path=bids_path)
> 
> ~/Development/mne-bids/mne_bids/read.py in read_raw_bids(bids_path, extra_params, verbose)
>     515                                             on_error='warn')
>     516     if channels_fname is not None:
> --> 517         raw = _handle_channels_reading(channels_fname, bids_fname, raw)
>     518 
>     519     # Try to find an associated electrodes.tsv and coordsystem.json
> 
> ~/Development/mne-bids/mne_bids/read.py in _handle_channels_reading(channels_fname, bids_fname, raw)
>     377         if not (ch_names_raw[-1] == 'STI 014' and
>     378                 ch_names_raw[:-1] == ch_names_tsv):
> --> 379             raise RuntimeError(msg)
>     380 
>     381     # Now we can do some work.
> 
> RuntimeError: Channels do not correspond between raw data and the channels.tsv file. For MNE-BIDS, the channel names in the tsv MUST be equal and in the same order as the channels in the raw data.
> 
> 64 channels in tsv file: "/tmp/bci-bids/sub-001/eeg/sub-001_task-RestEyesClosed_channels.tsv"
> 
>  --> ['FC5', 'Fc3.', 'Fc1.', 'Fcz.', 'Fc2.', 'Fc4.', 'Fc6.', 'C5..', 'C3..', 'C1..', 'Cz..', 'C2..', 'C4..', 'C6..', 'Cp5.', 'Cp3.', 'Cp1.', 'Cpz.', 'Cp2.', 'Cp4.', 'Cp6.', 'Fp1.', 'Fpz.', 'Fp2.', 'Af7.', 'Af3.', 'Afz.', 'Af4.', 'Af8.', 'F7..', 'F5..', 'F3..', 'F1..', 'Fz..', 'F2..', 'F4..', 'F6..', 'F8..', 'Ft7.', 'Ft8.', 'T7..', 'T8..', 'T9..', 'T10.', 'Tp7.', 'Tp8.', 'P7..', 'P5..', 'P3..', 'P1..', 'Pz..', 'P2..', 'P4..', 'P6..', 'P8..', 'Po7.', 'Po3.', 'Poz.', 'Po4.', 'Po8.', 'O1..', 'Oz..', 'O2..', 'Iz..']
> 
> 64 channels in raw file: "sub-001_task-RestEyesClosed_eeg.edf"
> 
>  --> ['Fc5.', 'Fc3.', 'Fc1.', 'Fcz.', 'Fc2.', 'Fc4.', 'Fc6.', 'C5..', 'C3..', 'C1..', 'Cz..', 'C2..', 'C4..', 'C6..', 'Cp5.', 'Cp3.', 'Cp1.', 'Cpz.', 'Cp2.', 'Cp4.', 'Cp6.', 'Fp1.', 'Fpz.', 'Fp2.', 'Af7.', 'Af3.', 'Afz.', 'Af4.', 'Af8.', 'F7..', 'F5..', 'F3..', 'F1..', 'Fz..', 'F2..', 'F4..', 'F6..', 'F8..', 'Ft7.', 'Ft8.', 'T7..', 'T8..', 'T9..', 'T10.', 'Tp7.', 'Tp8.', 'P7..', 'P5..', 'P3..', 'P1..', 'Pz..', 'P2..', 'P4..', 'P6..', 'P8..', 'Po7.', 'Po3.', 'Poz.', 'Po4.', 'Po8.', 'O1..', 'Oz..', 'O2..', 'Iz..']

### Workaround
In this specific case, enforcing a format conversion to BrainVision works:
```python
write_raw_bids(raw=raw, bids_path=bids_path, overwrite=True,
               format='BrainVision')
read_raw_bids(bids_path=bids_path)
```
Output:
```
<RawBrainVision | sub-001_task-RestEyesClosed_eeg.eeg, 64 x 9760 (61.0 s), ~68 kB, data not loaded>
```

### Expected behavior
Either we should fail early (during writing) or not at all. (I'm all for the latter: it should just work)

I think renaming (sanitizing, in our case) channels before writing the BIDS data is a totally sane use case and should be properly supported.

Since we've started moving away from the "sidecar metadata must match raw header metadata" paradigm, we probably can simply remove the check for matching channel names between the two, which is causing the exception here?

cc @sappelhoff @agramfort @jasmainak @adam2392

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #697

Please note that the same issue still exists for iEEG data in `dig._handle_electrodes_reading()`, but I don't fully understand the code and therefore currently don't dare to touch it. @adam2392 might want to give this one a shot 😇

cc @MaelysSolal 

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [ ] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -34,6 +34,7 @@ Enhancements
 
 - Some datasets out in the real world have a non-standard ``stim_type`` instead of a ``trial_type`` column in ``*_events.tsv``. :func:`mne_bids.read_raw_bids` now makes use of this column, and emits a warning, encouraging users to rename it, by `Richard Höchenberger`_ (:gh:`680`)
 - When reading data where the same event name or trial type refers to different event or trigger values, we will now create a hierarchical event name in the form of ``trial_type/value``, e.g. ``stimulus/110``  by `Richard Höchenberger`_ (:gh:`688`)
+- When reading data via :func:`mne_bids.read_raw_bids`, the channel names specified in the BIDS ``*_channels.tsv`` files now always take precedence (and to not need to match) the channel names stored in the raw files anymore, by `Richard Höchenberger`_ (:gh:`691`)
```
Comment (agramfort) [inline doc/whats_new.rst:37]:
```suggestion
- When reading data via :func:`mne_bids.read_raw_bids`, the channel names specified in the BIDS ``*_channels.tsv`` files now always take precedence (and do not need to match) the channel names stored in the raw files anymore, by `Richard Höchenberger`_ (:gh:`691`)
```

**Review by agramfort — COMMENTED** [review]
to me it's the right thing to do !

**Comment by sappelhoff** [comment]
> to me it's the right thing to do !

agreed, and eventually the validator should check that channel names in raw and BIDS are the same.

**Comment by agramfort** [comment]
validator would then be required to read EEG/MEG formats. It's too much I
think. For me
we should not enforce that the names are the same. We take the raw files as
they are and
the sidecar take precedence.

>

**Comment by hoechenberger** [comment]
> > to me it's the right thing to do !
> 
> agreed, and eventually the validator should check that channel names in raw and BIDS are the same.

Yes I'm confused here, @sappelhoff said he agrees with you, but then suggests something that would enforce the opposite :) So just to clear up the confusion: Do you think this PR is doing the right thing, @sappelhoff?

**Comment by sappelhoff** [comment]
Yes it's the right thing to take the sidecar files as the truth.

The validator checking that raw and sidecar info is aligned would be more of a help to dataset curators to get their story straight (less room for error). --> but that's future talk ... and as Alex says, it may even never happen because we have nobody developing dataset readers in JS.

**Pull request merged**